### PR TITLE
Changed rust_types.h to use typedef instead of macros

### PR DESCRIPTION
**Pull request by Zephia-Sero**

### CODE REVIEW

**Comment by LemonjamesD** [comment]
Why?

**Comment by Zephia-Sero** [comment]
why not

**Comment by LemonjamesD** [comment]
Macro works just better or the same

File: src/rust_types.h [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-#define u64 uint64_t
-#define i8 int8_t
-#define i16 int16_t
-#define i32 int32_t
-#define i64 int64_t
-#define usize size_t
-#define isize ssize_t
+// NOTE: float, double, and long double are *NOT* guaranteed to be
+// 32, 64, and 128 bits respectively!
+typedef float f32;
```
Comment (LemonjamesD) [inline src/rust_types.h:8]: Keep the defines and remove these typedefs

**Comment by Zephia-Sero** [comment]
for this case, sure, but macros are just find-and-replace strings. i prefer using typedef when defining types because it doesn't break when you try to:
```c
#define INTPTR_WRONG int*
typedef int *INTPTR_RIGHT;
INTPTR_WRONG a, b, c; // defines an int*a, and an int b, and int c
INTPTR_RIGHT a, b, c; // defines 3 int*'s
```

File: src/rust_types.h [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
-#define u32 uint32_t
-#define u64 uint64_t
-#define i8 int8_t
-#define i16 int16_t
-#define i32 int32_t
-#define i64 int64_t
-#define usize size_t
-#define isize ssize_t
+// NOTE: float, double, and long double are *NOT* guaranteed to be
+// 32, 64, and 128 bits respectively!
```
Comment (LemonjamesD) [inline src/rust_types.h:7]: Keep this comment

**Review by LemonjamesD — CHANGES_REQUESTED** [review]

**Comment by LemonjamesD** [comment]
> for this case, sure, but macros are just find-and-replace strings. i prefer using typedef when defining types because it doesn't break when you try to:
> 
> ```c
> #define INTPTR_WRONG int*
> typedef int *INTPTR_RIGHT;
> INTPTR_WRONG a, b, c; // defines an int*a, and an int b, and int c
> INTPTR_RIGHT a, b, c; // defines 3 int*'s
> ```

We're not using any pointers in the types so this doesn't apply

**Comment by Zephia-Sero** [comment]
yeah but a typedef is a type def. macro doesn't even have "type" in the name!

**Comment by LemonjamesD** [comment]
Pretty irrelevant will close unless you something else to say, this just seems like a very unnecessary change imo

**Comment by Zephia-Sero** [comment]
if it's a change that does nothing and at best just improves clarity by a minimal amount, and it's already been written, then what's the harm in including the change?

**Comment by LemonjamesD** [comment]
fair

**Review by LemonjamesD — APPROVED** [review]

**Pull request merged**